adapters/beachfront/beachfront.go

```diff
@@ -0,0 +1,486 @@
+	Site     BeachfrontSite        `json:"site"`
+	Device   BeachfrontVideoDevice `json:"device"`
+	User     openrtb.User          `json:"user"`
+	Cur      []string              `json:"cur"`
+}
+
+type BeachfrontSite struct {
+	Page string `json:"page"`
+}
+
+type BeachfrontPublisher struct {
+	Id string `json:"id"`
+}
+
+type BeachfrontVideoDevice struct {
+	Ua         string             `json:"ua"`
+	Devicetype int                `json:"deviceType"`
+	Geo        BeachfrontVideoGeo `json:"geo"`
+}
+
+type BeachfrontVideoGeo struct {
+	Ip string `json:"ip"`
+}
+
+type BeachfrontVideoImp struct {
+	Video    BeachfrontSize `json:"video"`
+	Bidfloor float64        `json:"bidfloor"`
+	Id       int            `json:"id"`
+	ImpId    string         `json:"impid"`
+}
+
+// ---------------------------------------------------
+//              Banner
+// ---------------------------------------------------
+type BeachfrontBannerRequest struct {
+	Slots          []BeachfrontSlot `json:"slots"`
+	Domain         string           `json:"domain"`
+	Page           string           `json:"page"`
+	Referrer       string           `json:"referrer"`
+	Search         string           `json:"search"`
+	Secure         int8             `json:"secure"`
+	DeviceOs       string           `json:"deviceOs"`
+	DeviceModel    string           `json:"deviceModel"`
+	IsMobile       int8             `json:"isMobile"`
+	Ua             string           `json:"ua"`
+	Dnt            int8             `json:"dnt"`
+	User           string           `json:"user"`
+	AdapterName    string           `json:"adapterName"`
+	AdapterVersion string           `json:"adapterVersion"`
+	Ip             string           `json:"ip"`
+}
+
+type BeachfrontSlot struct {
+	Slot     string           `json:"slot"`
+	Id       string           `json:"id"`
+	Bidfloor float64          `json:"bidfloor"`
+	Sizes    []BeachfrontSize `json:"sizes"`
+}
+
+type BeachfrontSize struct {
+	W uint64 `json:"w"`
+	H uint64 `json:"h"`
+}
+
+type BeachfrontResponseSlot struct {
+	CrID  string  `json:"crid"`
+	Price float64 `json:"price"`
+	W     uint64  `json:"w"`
+	H     uint64  `json:"h"`
+	Slot  string  `json:"slot"`
+	Adm   string  `json:"adm"`
+}
+
+func (a *BeachfrontAdapter) Name() string {
+	return "beachfront"
+}
+
+func (a *BeachfrontAdapter) SkipNoCookies() bool {
+	return true
+}
+
+func NewBeachfrontBannerRequest() BeachfrontBannerRequest {
+	r := BeachfrontBannerRequest{}
+	r.AdapterName = beachfrontAdapterName
+	r.AdapterVersion = beachfrontAdapterVersion
+
+	return r
+}
+
+func NewBeachfrontVideoRequest() BeachfrontVideoRequest {
+	r := BeachfrontVideoRequest{}
+	r.Cur = append(r.Cur, "USD")
+	r.IsPrebid = true
+
+	return r
+}
+
+func (a *BeachfrontAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+
+	var beachfrontRequests BeachfrontRequests
+	var reqJSON []byte
+	var uri string
+
+	uri = func() string {
```

Comment:
little strange... why the inline function & immediate execution here?

Seems like you should either move this into a named helper function, or just extract the logic so that `MakeRequests` runs it directly.